### PR TITLE
Add kotlin support

### PR DESCRIPTION
**Pull request by alachambre**

### CODE REVIEW

File: src/main/resources/archetype-resources/src/main/kotlin/__className__.kt [hunk 1]
```diff
@@ -4,52 +4,56 @@ import java.util.logging.Logger
...
+import org.bonitasoft.engine.api.APIAccessor
 
-class ${className} : AbstractConnector {
+class ${className} : AbstractUserFilter {
 
     constructor() : super()
 
-    val defaultInput = "defaultInput"
-    val defaultOutput = "defaultOutput"
+    companion object {
```
Comment (rbioteau) [inline src/main/resources/archetype-resources/src/main/kotlin/__className__.kt:14]: Is the usage of a companion object mandatory ?

File: src/main/resources/archetype-resources/src/main/kotlin/__className__.kt [hunk 2]
```diff
@@ -4,52 +4,56 @@ import java.util.logging.Logger
...
+        const val MAXIMUM_WORKLOAD_INPUT = "maximumWorkload"
+    }
+    
     val logger = Logger.getLogger(${className}::class.java.name)
 
     /**
-     * Perform validation on the inputs defined on the connector definition (src/main/resources/${artifactId}.def)
-     * You should: 
+     * Perform validation on the inputs defined on the actorfilter definition (src/main/resources/${artifactId}.def)
```
Comment (rbioteau) [inline src/main/resources/archetype-resources/src/main/kotlin/__className__.kt:21]:
```suggestion
     * Perform validation on the inputs defined on the actor filter definition (src/main/resources/${artifactId}.def)
```

File: src/test/resources/projects/testActorFilterKotlinProject/reference/pom.xml [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            </plugin>
+            <plugin>
+                <groupId>org.jetbrains.kotlin</groupId>
+                <artifactId>kotlin-maven-plugin</artifactId>
+                <version>${kotlin.version}</version>
+    
+                <executions>
+                    <execution>
+                        <id>compile</id>
+                        <goals> <goal>compile</goal> </goals>
```
Comment (rbioteau) [inline src/test/resources/projects/testActorFilterKotlinProject/reference/pom.xml:78]: format

File: src/main/resources/archetype-resources/src/main/kotlin/__className__.kt (hunk 1 above)
Comment (alachambre) [reply src/main/resources/archetype-resources/src/main/kotlin/__className__.kt:14]: Yes, it's the way to make a field 'static final' in kotlin. 
If we remove the companion then we cannot use 'const val' -> value will be on the instance and not the class. I did this for consistency with Java and Groovy.

Comment (alachambre) [reply src/main/resources/archetype-resources/src/main/kotlin/__className__.kt:14]: https://www.baeldung.com/kotlin-java-static-final

**Pull request merged**